picasso/src/main/java/com/squareup/picasso/Downloader.java

```diff
@@ -46,9 +45,9 @@
     final boolean localCacheOnly;
     final int responseCode;
 
-    public ResponseException(String message, boolean localCacheOnly, int responseCode) {
+    public ResponseException(String message, int networkPolicy, int responseCode) {
```

Comment:
For 2).

enum helps shape the API by disallowing any int being passed in. int is used internally to preserve object allocation by performing bitwise operations instead.